lib/Serialization/SerializeSIL.cpp

```diff
@@ -974,7 +978,108 @@ void SILSerializer::writeSILInstruction(const SILInstruction &SI) {
     break;
   }
 
-  case SILInstructionKind::DebugValueInst:
+  case SILInstructionKind::DebugValueInst: {
+    if (!SerializeDebugInfoSIL)
+     return;
+    auto DVI = cast<DebugValueInst>(&SI);
+    unsigned attrs = unsigned(DVI->poisonRefs() & 0x1);
+    attrs |= unsigned(DVI->usesMoveableValueDebugInfo()) << 1;
+    attrs |= unsigned(DVI->hasTrace()) << 2;
+
+    auto Operand = DVI->getOperand();
+    auto Type = Operand->getType();
+
+    unsigned DebugVarTypeCategory = 0;
+    auto DebugVar = DVI->getVarInfo();
+    const SILDebugScope *ScopeToWrite = nullptr;
+
+    auto &SM = SI.getFunction()->getModule().getSourceManager();
+
+    SmallVector<uint64_t, 8> ListOfValues;
+    ListOfValues.push_back(addValueRef(Operand));
+    ListOfValues.push_back(S.addTypeRef(Type.getRawASTType()));
+
+    if (DebugVar) {
+      attrs |= 1 << 3; // has debug var
+      attrs |= DebugVar->isLet() << 4;
+      attrs |= DebugVar->isDenseMapSingleton << 5; // needs two bits
+
+      ListOfValues.push_back(S.addUniquedStringRef(DebugVar->Name));
+      ListOfValues.push_back(DebugVar->ArgNo);
+
+      if (DebugVar->Type.has_value()) {
+        attrs |= 1 << 7;
+        ListOfValues.push_back(S.addTypeRef(DebugVar->Type->getRawASTType()));
+        DebugVarTypeCategory = (unsigned)DebugVar->Type->getCategory();
+      }
+
+      if (DebugVar->Scope) {
+        attrs |= 1 << 8;
+        ScopeToWrite = DebugVar->Scope;
+      }
+
+      if (DebugVar->Loc) {
+        auto RawLoc = DebugVar->Loc.value();
+        SourceLoc Loc = RawLoc.getSourceLoc();
+        if (Loc.isValid()) {
+          attrs |= 1 << 9;
+          auto LC = SM.getPresumedLineAndColumnForLoc(Loc);
+          auto FName = SM.getDisplayNameForLoc(Loc);
+          auto FNameID = S.addUniquedStringRef(FName);
+
+          ListOfValues.push_back(LC.first);
+          ListOfValues.push_back(LC.second);
+          ListOfValues.push_back(FNameID);
+        } else if (RawLoc.isFilenameAndLocation()) {
+          // TODO: this is a workaround until rdar://problem/25225083 is
```

Comment:
This comment isn't useful for Swift contributors without access to the link. Can you summarize what the problem is here?